### PR TITLE
Added EnvironmentalHazard events.

### PR DESCRIPTION
**Pull request by NaoUnderscore**

### CODE REVIEW

**Comment by IRacle1** [comment]
And how do you normally distinguish between Tantrum and Sinkhole in these events?
Maybe need to create an enum with `EnvironmentalHazard` types?

**Comment by NaoUnderscore** [comment]
> And how do you can distinguish between Tantrum and Sinkhole in these events? Maybe need to create an enum with `EnvironmentalHazard` types?

Sinkhole and Tantrum inherit from the base `EnvironmentalHazard` class.

**Comment by IRacle1** [comment]
> > And how do you can distinguish between Tantrum and Sinkhole in these events? Maybe need to create an enum with `EnvironmentalHazard` types?
> 
> Sinkhole and Tantrum inherit from the base `EnvironmentalHazard` class.

no, you don't understand me, for example we have `StayingOnEnvironmentalHazardEventArgs` , i (as plugin dev) want to check is a player stay on sinkhole.
How can I check in this event what exactly is the player standing on? Sinkhole or Tantrum?

**Comment by IRacle1** [comment]
🙃

**Review by Michal78900 — APPROVED** [review]

File: Exiled.Events/Patches/Events/Player/StayingOnTantrumEnvironmentalHazard.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                new(OpCodes.Ldc_I4_1),
+                new(OpCodes.Add),
+                new(OpCodes.Stloc_S, cnt.LocalIndex),
+                new CodeInstruction(OpCodes.Ldloc_S, cnt.LocalIndex).WithLabels(lpEnd),
+                new(OpCodes.Ldarg_0),
+                new(OpCodes.Callvirt, PropertyGetter(typeof(TantrumEnvironmentalHazard), nameof(TantrumEnvironmentalHazard.AffectedPlayers))),
+                new(OpCodes.Callvirt, PropertyGetter(typeof(List<ReferenceHub>), nameof(List<ReferenceHub>.Count))),
+                new(OpCodes.Clt),
+                new(OpCodes.Stloc_S, lpAlive.LocalIndex),
+                new(OpCodes.Ldloc_S, lpAlive.LocalIndex),
```
Comment (IRacle1) [inline Exiled.Events/Patches/Events/Player/StayingOnTantrumEnvironmentalHazard.cs:60]: You use `lpAlive` only here, do you need a that local var?

Comment (NaoUnderscore) [reply Exiled.Events/Patches/Events/Player/StayingOnTantrumEnvironmentalHazard.cs:60]: That's how for loops work in IL.

**Review by louis1706 — APPROVED** [review]
I have tested it work fine :)

**Review by IRacle1 — APPROVED** [review]
I've always wanted to click "Approve"

**Review by BuildBoy12 — APPROVED** [review]

**Pull request merged**